packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+
+The remediation includes the shared secret as both a QR Code and as text. It also includes a request to enter a Google Authenticator code after enrollment.
+
+The sign-on controller code for `handleEnrollAuthenticator(remdiation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
+
+```swift
+func handleEnrollAuthenticator(_ remdiation: Remediation) {
+    // Tell the caller to present a UI for enrolling in Google Authenticator.
+
+}
+```
+
+The collection of authenticators related to the current request are in the `authenticators` property of the remediation. Each one is an instance of the `Authenticator` class which includes an identifier in the `key` property and a name for display in the UI in the `displayName` property.
+
+Authenticators may include one or more capabilities that describe associated data and behaviors, such as the QR Code or the ability to send a code. This includes
+
+This utility function finds the Google Authenticator (`key` set to `"google_otp"`), and then loads an image for the QR Code and the shared secret:
+
+
+```swift
+// The information for constructing a UI for enrollment.
+typealias AuthenticatorEnrollment = (name: String?,
+                                     qrCode: UIImage?,
+                                     sharedSecret: String?)
+
+func authenticatorEnrollment() -> AuthenticatorEnrollment? {
+        guard let remediation = currentRemediation,
+              let authenticator = remediation.authenticators.filter({ $0.key == "google_otp"}).first,
+              let capability = authenticator.capabilities.filter({$0 is Capability.OTP}).first as? Capability.OTP,
+              let qrCode = capability.image,
+              let sharedSecret = capability.sharedSecret else {
+                  return nil
+              }
+
+        return (authenticator.displayName, qrCode, sharedSecret)
+    }
+```
+
+The model and view information for the one-time code entry field are nested in the forms of the remediation. First find a field with the `name` property set to `"credentials"` in the `form` property of the remediation. Then in the `form` property of that field, find the field with the `name` set to `"passcode"`:
+
+```swift
+typealias AuthenticatorCode = (codeField: Remediation.Form.Field,
+                               entryField: Remediation.Form.Field)
+
+    func authenticatorCodeField() -> AuthenticatorCode? {
+    guard let codeField = currentRemediation?["credentials"],
+          let entryField = currentRemediation?["credentials.passcode"] else {
+              return nil
+          }
+        return (codeField, entryField)
+    }
+
+```
+
+Set the `value` property of the  entry field, the one with a name `"passcode"` to the code entered by the user, and then submit the remediation to continue to the next step. This is the utility function from the sign-on manager:
```

Comment:
```suggestion
Set the `value` property of the  entry field, the one with a name `"passcode"` to the OTP entered by the user, and then submit the remediation to continue to the next step. This is the utility function from the sign-on manager:
```

Reply:
I modified per above and added a definition of OTP elsewhere (per the style guide on Acronyms and abbreviations). I know that TOTP was defined earlier in the guide, but it seems that they are two different acronyms, even though the OTP part is the same (but can we be sure the developer knows this.) I defined the acronym in the paragraph on roughly line 351 starting "Authenticators may include..." I modified the end of the first sentence to:

> or the ability to send a one-time passcode (OTP).

Reply:
I also updated the other occurrences where "code" was used instead of OTP. It begs the question of modifying the source examples as well.

Reply:
As long as we are consistent in the guide, I think that's key. Occasionally the SDK \ Samples still use things like Log In and "code" vs "otp" is another one of those inconsistencies. I'm not sure when an OTP is Time-based or not. I think they always are, but sometimes they are more apparently so during challenge flows as they keep changing in the authenticator apps but not so much when used for enrollment such as with GoogleAuth. Which is why I suggest OTP rather than TOTP if you're curious.

Reply:
FYI, I agree with using OTP... Google Authenticator is a TOTP, so the OTP is time-based, but there [are other types](https://www.protectimus.com/blog/otp-generation-algorithms-and-token-types/).

Reply:
I think this one is closed right?

Reply:
yup. Closed